### PR TITLE
Add rich responses for Telegram-CX Integration

### PR DESCRIPTION
**Pull request by leulaf**
Adds support for rich responses such as images, audios and buttons.

### CODE REVIEW

File: cx/telegram/server.js [hunk 1]
```diff
@@ -31,64 +32,78 @@ const {SessionsClient} = require('@google-cloud/dialogflow-cx');
...
+        let reply;
+
+        switch(true) {
+            case response.hasOwnProperty('text'): {
+                reply = {
+                    chat_id: chatId,
+                    text: response.text.text.join()
+                };  
+            }
+            break;
```
Comment (jrash06) [inline cx/telegram/server.js:64]: I think it's better to put break; into {...} block

File: cx/telegram/server.js [hunk 2]
```diff
@@ -31,64 +32,78 @@ const {SessionsClient} = require('@google-cloud/dialogflow-cx');
...
+             * The layout for the custom payload responses can be found in these sites:
+             * Buttons: https://core.telegram.org/bots/api#inlinekeyboardmarkup
+             * Photos: https://core.telegram.org/bots/api#sendphoto
+             * Voice Audios: https://core.telegram.org/bots/api#sendvoice
+             */
+            case response.hasOwnProperty('payload'): {
+                reply = await structProtoToJson(response.payload)
+                reply['chat_id'] = chatId
+            }
+            break;
```
Comment (jrash06) [inline cx/telegram/server.js:76]: Ditto

File: cx/telegram/server.js [hunk 3]
```diff
@@ -31,64 +32,78 @@ const {SessionsClient} = require('@google-cloud/dialogflow-cx');
...
-    agentId,
-    sessionId
-  );
-  console.info(sessionPath);
-  
-  request = telegramToDetectIntent(telegramRequest, sessionPath);
-  const [response] = await client.detectIntent(request);
-
-  return response;
+    const sessionId = telegramRequest.message.chat.id;
```
Comment (jrash06) [inline cx/telegram/server.js:94]: shall we observe the original 2-space indent, here and anywhere?

I see you're using 4-space indent

File: cx/telegram/server.js (hunk 1 above)
Comment (leulaf) [reply cx/telegram/server.js]: Restructured the code.

File: cx/telegram/server.js (hunk 2 above)
Comment (leulaf) [reply cx/telegram/server.js]: Restructured the code.

**Review by tyhu-google — APPROVED** [review]

File: cx/telegram/server.js (hunk 3 above)
Comment (leulaf) [reply cx/telegram/server.js:94]: I have used 4-space indent through out the integration files. The original was just autoformated to a 2-space indent by accident for a couple of lines.

Comment (chihhsuan-google) [reply cx/telegram/server.js:94]: Just to confirm, there will be another PR to auto-format and fix the style?

Comment (leulaf) [reply cx/telegram/server.js:94]: Yes, I will be creating another PR to format and fix the style.

File: cx/telegram/README.md [hunk 4]
```diff
@@ -42,3 +43,15 @@ More information can be found in Cloud Run
...
+### 
+**Testing the Integration**
+
+*   You have now completed all the necessary steps and can test your integration.
+*   To test rich messages you have to first input in the [custom payload](https://cloud.google.com/dialogflow/cx/docs/concept/fulfillment) for your agent:
+    *   To do this first click "Add dialogue option" in a fulfillment editing panel.
+    *   Then copy and paste one of the bottom codes and edit it to fit your needs.
+        *   { "photo": "http://example.com/image.jpg" } to [send photos](https://core.telegram.org/bots/api#sendphoto).
+        *   { "voice": "http://example.com/voice.mp3" } to [send voice audios](https://core.telegram.org/bots/api#sendvoice).
+        *   {"text": "Test Text", "reply_markup": {"inline_keyboard": [ [ { ... } ] ] } to [send buttons](https://core.telegram.org/bots/api#inlinekeyboardmarkup)
```
[the remark this answers is not in the document]
Comment (leulaf) [reply cx/telegram/README.md]: Fixed the syle to make it more consistent.

**Comment by chihhsuan-google** [comment]
Thanks for the work, Leul!

**Pull request merged**